app/positiondirection.h

```diff
@@ -0,0 +1,58 @@
+#ifndef POSITIONDIRECTION_H
+#define POSITIONDIRECTION_H
+
+#include <QObject>
+#include <QTimer>
+
+#ifdef MOBILE_OS
+#include <QOrientationSensor>
+#include <QCompass>
+#endif
+
+#include "qgsquickpositionkit.h"
+
+class PositionDirection : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY( qreal direction READ direction NOTIFY directionChanged )
+    Q_PROPERTY( QgsQuickPositionKit *positionKit READ positionKit WRITE setPositionKit NOTIFY positionKitChanged )
+    Q_PROPERTY( const qreal MIN_INVALID_DIRECTION READ minInvalidDirection NOTIFY minInvalidDirectionChanged )
+
+  public:
+    explicit PositionDirection( QObject *parent = nullptr );
+
+    qreal direction() const;
+
+    QgsQuickPositionKit *positionKit() const;
+    void setPositionKit( QgsQuickPositionKit *positionKit );
+
+    qreal minInvalidDirection() const;
+
+  signals:
+    void directionChanged();
+    void positionKitChanged();
+    void minInvalidDirectionChanged();
+  public slots:
+#ifdef MOBILE_OS
+    void updateDirection();
+    void setUserOrientation();
+#endif
+
+  private:
+    //! any direction value in degrees that is < -180 is not valid.
+    qreal MIN_INVALID_DIRECTION = -180.1;
+    qreal mDirection = MIN_INVALID_DIRECTION;
+    QgsQuickPositionKit *mPositionKit = nullptr;
+#ifdef MOBILE_OS
```

Comment:
Is it possible to get it working without these #ifdefs for MOBILE_OS ? Is it that some platforms do not have QOrientationSensor and QCompass classes defined at all? If they just don't return sensible results, we could just keep the sensor/compass as null pointers and not start the timer...

Reply:
QOrientationSensor and QCompass are from Qt.Sensors which have been added only to iOS and Android build. The Win build is kept untouched (cannot test it anyway) as well as desktop. We can add Qt.Sensors to all builds and make a code more OS independent (with ignoring functionality if sensors are not available). Dont have strong idea on this.